src/Cohesity.Powershell/Scripts/Routes/Remove-CohesityRoutes.ps1

```diff
@@ -0,0 +1,74 @@
+function Remove-CohesityRoutes {
+    <#
+        .SYNOPSIS
+        Removes the routes.
+        .DESCRIPTION
+        The Remove-CohesityRoutes function is used to remove routes.
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Remove-CohesityRoutes -DestNetwork "10.2.3.4" -NextHop "10.2.3.5" -InterfaceGroupName "intf_group1"
+        .EXAMPLE
+        Get-CohesityRoutes -FilterName INTERFACE-GROUP-NAME -FilterValue "intf_group1" | Remove-CohesityRoutes
+        .EXAMPLE
+        Get-CohesityRoutes -FilterName DESTINATION-NETWORK -FilterValue "1.2.4.14/32" | Remove-CohesityRoutes
+    #>
+    [CmdletBinding(SupportsShouldProcess = $True, ConfirmImpact = "High", DefaultParameterSetName='Default')]
+    Param(
+        [Parameter(Mandatory = $true, ParameterSetName = 'NonPiped')]
+        [ValidateNotNullOrEmpty()]
+        $DestNetwork,
```

Comment:
As these parameter are mandatory, won't it ask for a cmdline input from the user, in case of using the pipeline?

Reply:
It would not ask for mandatory parameters when used in conjunction with ParameterSetName

Reply:
can you group these parameters then ? I remember powershell allows grouping. 
You can add routeObject to one group and rest of them in other.

Reply:
Could not find any keyword that does parameterset grouping. Any pointers?

Reply:
parametersetname doesnt make it mandatory ?

Reply:
ParameterSetName is the keyword to use for grouping. In this case DestNetwork, NextHop and InterfaceGroupName are grouped together with ParameterSetName=NonPiped, and RouteObject is grouped in ParameterSetName=Default.

ParameterSetName not necessarily makes the parameter mandatory.